make_prg/subcommands/update.py

```diff
@@ -0,0 +1,187 @@
+import multiprocessing
+import os
+from pathlib import Path
+from loguru import logger
+from make_prg.utils import io_utils, gfa
+from make_prg.update.denovo_variants import DenovoVariantsDB
+from make_prg.prg_builder import PrgBuilderZipDatabase, LeafNotFoundException
+from make_prg.utils.msa_aligner import MAFFT, MSAAligner
+from dataclasses import dataclass
+
+
+def register_parser(subparsers):
+    subparser_update_prg = subparsers.add_parser(
+        "update",
+        usage="make_prg update",
+        help="Update PRGs given new sequences.",
+    )
+
+    def check_if_is_update_file(argument: str):
+        argument = Path(argument)
+        is_zip_file = argument.suffix == ".zip"
+        is_update_DS = argument.suffix == ".update_DS"
+        if not is_zip_file and not is_update_DS:
+            subparser_update_prg.error(f"{argument} is not a update_DS nor a zip file.")
+        return argument
+    subparser_update_prg.add_argument(
+        "-u",
+        "--update_DS",
+        action="store",
+        type=lambda argument: check_if_is_update_file(argument),
+        required=True,
+        help=(
+            "Filepath to the update data structures (a *.update_DS.zip file created "
+            "by make_prg from_msa)"
+        ),
+
+    )
+    subparser_update_prg.add_argument(
+        "-d",
+        "--denovo_paths",
+        action="store",
+        type=str,
+        required=True,
+        help=(
+            "Filepath containing denovo sequences. Should point to a denovo_paths.txt file"
+        ),
+    )
+    subparser_update_prg.add_argument(
+        "-o",
+        "--output_prefix",
+        action="store",
+        type=str,
+        required=True,
+        help="Prefix for the output files",
+    )
+    subparser_update_prg.add_argument(
+        "-t",
+        "--threads",
+        action="store",
+        type=int,
+        default=1,
+        help="Number of threads (default 1)",
+    )
+    subparser_update_prg.add_argument(
+        "--mafft",
+        help="Path to MAFFT executable. By default, it is assumed to be on $PATH",
+        default="mafft",
+    )
+    subparser_update_prg.add_argument(
+        "--output_graphs",
+        dest="output_graphs",
+        action="store_true",
+        default=False,
+        help="Outputs the recursive tree graphical representation (for development use only)",
+    )
+    subparser_update_prg.set_defaults(func=run)
+
+    return subparser_update_prg
+
+
+@dataclass
```

Comment:
Not sure if you know, but dataclasses were only added in python 3.8. Are we raising our minimum supported python version?

An alternative is to use named tuples like https://github.com/mbhall88/MemoryUnits/blob/18551958dbf51a9f68e8733ba17b880dbecc0da7/memory_units.py#L18

Reply:
Oooops, I overlooked this. In the previous PR there was a suggestion to change some classes into `dataclasses`, and I really liked it, and changed a few classes throughout the code to `dataclasses`, but did not pay attention at the minimum python version. I think this is a broader discussion, so I am also inviting @bricoletc here. My opinion is that I would be fine raising our minimum supported python version to `3.7`, as it is fairly standard now, although some systems come preinstalled with only `3.6`. Not sure about raising to `3.8` though, I would like to collect both your opinions, and then we decide.

TODO (not for myself): if we decide to not keep `dataclasses` in order to decrease our minimum supported python version to `3.6` or `3.7`, change the github workflow CI/CD image to use the version we target

Reply:
I'm fine to have 3.8 as minimum. 3.6 is now end of life anyway.

Reply:
I'm fine to have 3.8 as minimum too now

Reply:
yep i agree though do we say this somewhere e.g. the README?

Reply:
Not yet, will add it to dependencies section in Readme, but Readme updates will come in a future PR only.

Reply:
Closing, feel free to reopen